filesystem/filesystem.go

```diff
@@ -49,22 +57,52 @@ func getFileSystemInfo() (interface{}, error) {
 }
 
 func parseDfOutput(out string) ([]interface{}, error) {
-	var aggregateError error
 	lines := strings.Split(out, "\n")
 	if len(lines) < 2 {
 		return nil, errors.New("no output")
 	}
 	var fileSystemInfo = make([]interface{}, 0, len(lines)-2)
+
+	// parse the header to find the offsets for each component we need
+	hdr := lines[0]
+	fieldErrFunc := func(field string) error {
+		return fmt.Errorf("could not find '%s' in `%s %s` output",
+			field, dfCommand, strings.Join(dfOptions, " "))
+	}
+
+	kbsizeOffset := strings.Index(hdr, "1K-blocks")
```

Comment:
I don't think we can reliably key off from the header. The [strings in the command](https://github.com/coreutils/coreutils/blob/master/src/df.c#L180-L183) are [wrapped in `N_()` macro](https://github.com/coreutils/coreutils/blob/master/src/df.c#L180-L183) and [some text](https://github.com/coreutils/coreutils/blob/master/src/df.c#L620-L632) is wrapped in [`_()` macro](https://www.freedesktop.org/software/gstreamer-sdk/data/docs/latest/glib/glib-I18N.html#gettext-macro) which [might be implying](https://www.freedesktop.org/software/gstreamer-sdk/data/docs/latest/glib/glib-I18N.html#N-:CAPS) that there may be i10n around them.

Because of this, I wonder if we should ensure we use `-p` and/or `--output` and not depend on headers to parse this data. As a follow-up, do we maybe have some newer golang primitives for this too so that we don't have to deal with CLI?

Reply:
Ah, good point.  I think we'd need to set some localization variables like `LC_ALL=C` to ensure we get the un-translated output.

Using `-P` or `--output` would have the same problem -- different headers, but still variable-width.  And switching to gopsutil might be a breaking change, in terms of which filesystems are included or excluded.  We really don't have enough data to tell.